### PR TITLE
fix(request-validator): fix content-type breaking change

### PR DESCRIPTION
**Pull request by vm-001**
### Description

What did you change and why?
 
Include any supporting resources, e.g. link to a Jira ticket, GH issue, FTI, Slack, Aha, etc.

### Testing instructions

Netlify link: 

### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

File: app/_hub/kong-inc/request-validator/_index.md
Comment (lena-larionova) [inline app/_hub/kong-inc/request-validator/_index.md:37]: I'm confused. Which release are all these changes going into, exactly?

Comment (vm-001) [reply app/_hub/kong-inc/request-validator/_index.md:37]: There are a couple of things

- content-type with parameter was introduced at v2.8 and > v3.1
- fixing breaking behavior at v2.8
- 3.0 is a gap(for some reason) that as same as version < v2.7  (I found the v3.0.x is showing the 2.8.x's description, it seems like the (minimum_version: "3.0.x"  and maximum_version: "3.0.x" does not work)

Comment (lena-larionova) [reply app/_hub/kong-inc/request-validator/_index.md:37]: Very weird. Luckily, we'll have this type of thing under control soon, as we're switching to generating these docs from plugin schemas in the near future :) Thanks for the fixes!

File: app/_hub/kong-inc/request-validator/_index.md
Comment (vm-001) [inline app/_hub/kong-inc/request-validator/_index.md:95]:
```suggestion

```

File: app/_hub/kong-inc/request-validator/_index.md
Comment (vm-001) [reply app/_hub/kong-inc/request-validator/_index.md:37]: LGTM. I made another small change.

**Comment by vm-001** [comment]
Holding this doc PR until the code PR merged. https://github.com/Kong/kong-ee/pull/4808

**Review by samugi — CHANGES_REQUESTED** [review]
Please see my comment [here](https://github.com/Kong/kong-ee/pull/4808#discussion_r1173462181)

**Comment by vm-001** [comment]
@samugi @lena-larionova Could you please take another look?

File: app/_hub/kong-inc/request-validator/_index.md
Comment (samugi) [inline app/_hub/kong-inc/request-validator/_index.md:50]: This bit is only visible "from 3.1.x" so probably not necessary to include the version in this note.

**Review by samugi — APPROVED** [review]
The change I requested looks good to me now, I'll let @lena-larionova comment about any improvements on wording.

File: app/_hub/kong-inc/request-validator/_index.md
Comment (lena-larionova) [reply app/_hub/kong-inc/request-validator/_index.md:50]: Agreed. I'll adjust it.

Comment (lena-larionova) [inline app/_hub/kong-inc/request-validator/_index.md:50]:
```suggestion
        * The parameter is checked for an exact match, which means a request with a parameter (e.g. `application/json; charset=UTF-8`) is **NOT** considered valid for one without the same parameter (e.g. `application/json`).
```
How's this? "The parameter is checked forcibly" is slightly awkward phrasing.

**Review by lena-larionova — COMMENTED** [review]
Please check my suggested change to make sure it's still accurate.

File: app/_hub/kong-inc/request-validator/_index.md
Comment (vm-001) [reply app/_hub/kong-inc/request-validator/_index.md:50]: Using “exact match” may not be accurate since the charset parameter was treated as case-insensitive. 
ChatGPT told me another phase is  "the parameter is strictly validated"

Comment (lena-larionova) [reply app/_hub/kong-inc/request-validator/_index.md:50]: I like it. Thanks, ChatGPT 😁

**Comment by lena-larionova** [comment]
@vm-001 Almost done, one last thing: looks like we don't need to wait to release this with 3.3, so can you target this PR to main? Then we can merge.

**Review by lena-larionova — APPROVED** [review]

**Pull request merged**
